pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -191,6 +220,15 @@ func (w *Watcher) init() error {
 // Goroutines started here will be waited for in Stop() before cleaning up.
 // Ignore all errors except root dir not being walkable
 func (w *Watcher) traversePluginDir(dir string) error {
```

Comment:
would it be simpler to:
1. have traversePluginDir call `w.handleCreateEvent` directly and synchronously, rather than enqueuing synthetic create events in the event channel
2. wait until the two traversePluginDir() calls have completed in Start() before starting the event processing goroutine

that would ensure that while traversing a given directory (either in Start() or in response to an observed directory creation), no observed events from the started watchers were processed

Reply:
Done